src/ast/passes/semantic_analyser.cpp

```diff
@@ -2930,7 +2939,11 @@ void SemanticAnalyser::visit(Tuple &tuple)
 
 void SemanticAnalyser::visit(ExprStatement &expr)
 {
-  visit(expr.expr);
+  if (auto *call = expr.expr.as<Call>()) {
+    visit_call(*call, true);
+  } else {
+    visit(expr.expr);
+  }
```

Comment:
Wouldn't it be easier to do
```suggestion
  if (auto *call = expr.expr.as<Call>()) {
    call->ret_val_discarded = true;
  }
  visit(expr.expr);
```
and no more modifications to the current code would be necessary?

Also a comment would be good as it took me some time to figure out why this is correct.

Reply:
Ha yeah, that's better.

Reply:
> Also a comment would be good as it took me some time to figure out why this is correct.

Done.